### PR TITLE
[09] Add confidence intervals to cross correlation lag plots

### LINKED ISSUE #9: Add confidence intervals to cross correlations

### PR DESCRIPTION
**Pull request by Jack-0-0**
Closes #9.

 This PR:
* Adds confidence intervals into the `lagplot` function at `iss_forecasting/analysis/utils/plotting.py`
* Adds updated cross correlation lagplots into  notebook at `iss_forecasting/analysis/tech_category_level/research_funding_and_investment.py` and adds a note on the confidence intervals of the plots

### CODE REVIEW

File: iss_forecasting/analysis/utils/plotting.py [hunk 1]
```diff
@@ -89,7 +90,9 @@ def lagplot(
     """
     x_ = x.shift(lag)
     y_ = y if y is not None else x
-    corr = y_.corr(x_)
+    pg_corr = pg.corr(x_, y_).round(2)
```
Comment (beingkk) [inline iss_forecasting/analysis/utils/plotting.py]: Which correlation coefficient method do you think would work best here?
Looks like the default is `pearson`, which however has a normality assumption.

**Comment by beingkk** [comment]
Hi, thanks for this - I'll try running it by end of tomorrow or Friday. Some additional quick thoughts below.

While thinking about the correlations, I'm wondering what's the most appropriate way to measure them in our case (as there are so many methods to get the coefficients and confidence intervals).

In our case, I'd say the main point of using correlations is simply to flag sectors where we might have something interesting going on - and then we can always sense-check by looking at the plots (and later by digging deeper into the organisations). So, I think the approach taken so far is perfectly fine.

In addition, it's might also be worth considering a bit the strengths and weakness of the methods, and be wary some potential pitfalls. For example, I think the package you've used calculates confidence intervals using Fisher transformation - are there cases where it might provide spurious results? (maybe [this paper](https://link.springer.com/article/10.3758/s13428-016-0702-8) has some insight)

The examples [in this discussion](https://stats.stackexchange.com/questions/133155/how-to-use-pearson-correlation-correctly-with-time-series) are also quite interesting, on the potential pitfalls when measuring correlations.

Don't think this necessitates changes to this PR, but just something to think about in parallel with the rest of the analyses.

File: iss_forecasting/analysis/utils/plotting.py (hunk 1 above)
Comment (Jack-0-0) [reply iss_forecasting/analysis/utils/plotting.py]: Thanks for spotting this. I have changed the method to `spearman`.

**Comment by beingkk** [comment]
Thanks, this is great - really helpful to see which correlations actually have confidence intervals that don't include `r = 0`. I think it's good to merge!

Looks like the general rule is that there is no significant correlation at the sector level, with the exception of a few cases, for example: `Heat storage` at lags 0, 2, and 4; `Low carbon heating` at lag 1 and lag 4; `EEM` lags 4 and 6, `Batteries`... perhaps it's good to write a note somewhere about these examples.

**Comment by Jack-0-0** [comment]
> Thanks, this is great - really helpful to see which correlations actually have confidence intervals that don't include `r = 0`. I think it's good to merge!
> 
> Looks like the general rule is that there is no significant correlation at the sector level, with the exception of a few cases, for example: `Heat storage` at lags 0, 2, and 4; `Low carbon heating` at lag 1 and lag 4; `EEM` lags 4 and 6, `Batteries`... perhaps it's good to write a note somewhere about these examples.

Thanks Karlis, I've added a note relating to this into the notebook.

**Pull request merged**